addons/knowledge/controllers/main.py

```diff
@@ -0,0 +1,319 @@
+        return self.get_tree_panel(article, 'knowledge.knowledge_article_tree', unfolded_articles=unfolded_articles)
+
+    @http.route('/knowledge/tree_panel/portal', type='json', auth='public')
+    def get_tree_panel_portal(self, active_article_id=False, unfolded_articles=False):
+        article = request.env["knowledge.article"].sudo().browse(active_article_id).exists()
+        return self.get_tree_panel(article, 'knowledge.knowledge_article_tree_frontend', unfolded_articles=unfolded_articles)
+
+    @http.route('/knowledge/tree_panel/children', type='json', auth='user')
+    def get_tree_panel_children(self, parent_id):
+        Article = request.env["knowledge.article"]
+        parent = Article.browse(parent_id).exists()
+        if not parent:
+            return werkzeug.exceptions.NotFound()
+        root_articles = Article.search([("parent_id", "=", parent_id)]).sorted('sequence')
+        return request.env['ir.qweb']._render('knowledge.articles_template', {
+            'articles': root_articles,
+            'readonly_mode': True,  # used to bypass access check (to speed up loading)
+        })
+
+    @http.route('/knowledge/tree_panel/favorites', type='json', auth='user')
+    def get_tree_panel_favorites(self, active_article_id=False):
+        favorite_articles = request.env['knowledge.article.favorite'].search([
+            ("user_id", "=", request.env.user.id), ('article_id.active', '=', True)
+        ])
+        return request.env['ir.qweb']._render('knowledge.knowledge_article_tree_favorites', {
+            'favorites': favorite_articles,
+            "active_article_id": active_article_id
+        })
+
+    # ------------------------
+    # Article permission panel
+    # ------------------------
+
+    @http.route('/knowledge/get_article_permission_panel_data', type='json', auth='user')
+    def get_article_permission_panel_data(self, article_id):
+        """
+        Returns a dictionnary containing all values required to render the permission panel.
+        :param article_id: (int) article id
+        """
+        Article = request.env['knowledge.article']
+        article = Article.browse(article_id)
+        if not article.user_has_access:
+            return werkzeug.exceptions.Forbidden()
+        is_sync = not article.is_desynchronized
+        # Get member permission info
+        members_values = []
+        members_permission = article._get_article_member_permissions()[article.id]
+
+        # retrieve all involved partners info
+        results = request.env['res.partner'].search_read([('id', 'in', list(members_permission.keys()))],
+                                                         fields=['id', 'name', 'user_ids', 'partner_share'])
+        partners_info = {
+            partner['id']: {'name': partner['name'], 'user_ids': partner['user_ids'], 'share': partner['partner_share']}
+            for partner in results}
```

Comment:
Wild suggestion:

How about updating the members_permission dict to use as a "main source of data"?

```py
for member_partner in members_partners:
    # update the "members_permission" dict to include the partner information
    member_permission = members_permission[member_partner['id']]
    member_permission['partner_name'] = member_partner['name']
    member_permission['partner_user_ids'] = member_partner['user_ids']
    member_permission['partner_share'] = member_partner['partner_share']
```